File: src/main/java/mods/eln/sim/electrical/heater/ResistorHeatThermalLoad.java
package mods.eln.sim.electrical.heater;

import mods.eln.sim.IProcess;
import mods.eln.sim.thermal.ThermalLoad;
import mods.eln.sim.electrical.mna.component.Resistor;

public class ResistorHeatThermalLoad implements IProcess {

    Resistor r;
    ThermalLoad load;

    public ResistorHeatThermalLoad(Resistor r, ThermalLoad load) {
        this.r = r;
        this.load = load;
    }

    @Override
    public void process(double time) {
        load.movePowerTo(r.getP());
    }
}

File: src/main/java/mods/eln/sim/thermal/ResistorHeatThermalLoad.kt
package mods.eln.sim.thermal

import mods.eln.sim.core.IProcess
import mods.eln.sim.thermal.ThermalLoad
import mods.eln.sim.mna.passive.Resistor

class ResistorHeatThermalLoad(internal var r: Resistor, internal var load: ThermalLoad) : IProcess {

    override fun process(time: Double) {
        load.movePowerTo(r.getPower())
    }
}
